resource/opentelemetry-resource-detector-container/tests/test_container.py

```diff
@@ -0,0 +1,120 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest.mock import mock_open, patch
+
+from opentelemetry import trace as trace_api
+from opentelemetry.resource.detector.container import ContainerResourceDetector
+from opentelemetry.sdk.resources import get_aggregated_resources
+from opentelemetry.semconv.resource import ResourceAttributes
+from opentelemetry.test.wsgitestutil import WsgiTestBase
+
+MockContainerResourceAttributes = {
+    ResourceAttributes.CONTAINER_ID: "7be92808767a667f35c8505cbf40d14e931ef6db5b0210329cf193b15ba9d605",
+}
+
+
+def simple_wsgi(environ, start_response):
+    start_response("200 OK", [("Content-Type", "text/plain")])
+
+    def response():
+        yield b"*"
+
+    return response()
+
+
+class ContainerResourceDetectorTest(WsgiTestBase):
+    @patch(
+        "builtins.open",
+        new_callable=mock_open,
+        read_data=f"""14:name=systemd:/docker/{MockContainerResourceAttributes[ResourceAttributes.CONTAINER_ID]}
```

Comment:
Does this data stand for v1 or v2?
Can you add a test for the other one and not just mocks?

Reply:
It should be viable for V1 or V2 anyone.

> Can you add a test for the other one and not just mocks?
What do you mean by test for the other one?

Reply:
I saw that you expected "hostname" in the data of v2, and I didn't see it in the read_data
Maybe I just didn't get it,
Can you please explain what's the differences between v1 and v2?

Reply:
yes thats what I meant. v1 and v2 get the container.id from different files and parsing is different as well. Thats why

Reply:
I don't get it. There are two different implementations for v1 and v2, but this mock doesn't contain anything for v2 where the "hostname" string is searched in the line. How can it be sufficient for both versions?

Reply:
I have made changes to test cases and removed the mock file.
for docker based containers you will find container.id in `mountinfo` file but for other containers hat information is present in cgroups file. thats why I have v1 and v2 implementation.

I have added the test case for the v2 to search for hostname or containers string